src/ui/DynamicFacet/DynamicFacetRange.ts

```diff
@@ -49,16 +49,27 @@ export class DynamicFacetRange extends DynamicFacet implements IComponentBinding
       localizedString: () => l('To'),
       section: 'CommonOptions'
     }),
+    field: ComponentOptions.buildFieldOption({ required: true, section: 'CommonOptions' }),
     /**
      * The string format to apply to the minimum and maximum value of each range displayed in the facet.
      *
-     * **Default:** [`number`]{@link DynamicFacetRangeValueFormat.number}
+     * **Default:** If the [`field`]{@link DynamicFacet.options.field} contains "date", the format will be [`date`]{@link DynamicFacetRangeValueFormat.date}.
+     * Else, it will be [`number`]{@link DynamicFacetRangeValueFormat.number}.
      *
      * @examples date
      */
     valueFormat: ComponentOptions.buildStringOption({
```

Comment:
You can pass a type when you build the string option eg: `buildStringOption<'number'|'date'>({...})`  and it will generate the proper "constraint" in the documentation.

Reply:
could it be `ComponentOptions.buildStringOption<DynamicFacetRangeValueFormat>`

Reply:
Yes, enums should work. When the docgen.json is generated, it's important that the last line in the constrained values array for the option has all the possible values. These are used by the IEditor's intellisense.